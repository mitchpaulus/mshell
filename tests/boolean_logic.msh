[(true false and) ("truey") ("falsey")] if wl
[(true true and) ("truey") ("falsey")] if wl
[(false false and) ("truey") ("falsey")] if wl
[(true false or) ("truey") ("falsey")] if wl
[(true true or) ("truey") ("falsey")] if wl
[(false false or) ("truey") ("falsey")] if wl

"" wl
"Nots" wl
"========" wl
[(true false and not) ("truey") ("falsey")] if wl
[(true true and not) ("truey") ("falsey")] if wl
[(false false and not) ("truey") ("falsey")] if wl
[(true false or not) ("truey") ("falsey")] if wl
[(true true or not) ("truey") ("falsey")] if wl
[(false false or not) ("truey") ("falsey")] if wl
