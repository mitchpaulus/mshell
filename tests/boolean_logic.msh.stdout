falsey
truey
falsey
truey
truey
falsey

Nots
========
truey
falsey
truey
falsey
falsey
truey

Not Equals
========

Trues
========
truey
truey

Falses
========
falsey
falsey
falsey

Short-circuiting
========
Should print
falsey
falsey
truey
Should print
falsey
